src/TestFramework/TestFramework/Logger.cs

```diff
@@ -40,13 +40,12 @@ public static void LogMessage(string format, params object?[] args)
             ? format
             : string.Format(CultureInfo.InvariantCulture, format, args);
 
-        object?[] parameters = [message];
         // Making sure all event handlers are called in sync on same thread.
-        foreach (Delegate invoker in OnLogMessage.GetInvocationList())
+        foreach (LogMessageHandler invoker in DelegatePolyfill.EnumerateInvocationList(OnLogMessage))
```

Comment:
Note: we don't call Logger.LogMessage **at all**. I don't know how much consumers will be calling this, but it doesn't seem to be a hot path.

@Evangelink I'm not sure if there are "valid" use cases for this Logger class or not. If not, we should consider breaking in v4.